src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -663,6 +663,16 @@
   develop(bool, VerifyAliases, false,                                       \
           "perform extra checks on the results of alias analysis")          \
                                                                             \
+  develop(uint, VerifyConstraintCasts, 0,                                   \
```

Comment:
Any downsides in making the flag diagnostic? It'll make it available in product builds.

Reply:
That's a great idea, done it.